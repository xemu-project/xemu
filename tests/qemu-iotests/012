#!/usr/bin/env bash
# group: auto quick
#
# Make sure we can open read-only images
#
# Regression fixed by commit 11a1feb6552e3a4709e454faea5e3be5ca8d9e6a.
#
# Copyright (C) 2009 Red Hat, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

# creator
owner=hch@lst.de

seq=`basename $0`
echo "QA output created by $seq"

status=1	# failure is the default!

_cleanup()
{
	_cleanup_test_img
}
trap "_cleanup; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_supported_fmt generic
_supported_proto file

# Remove once all tests are fixed to use TEST_IMG_FILE
# correctly and common.rc sets it unconditionally
test -z "$TEST_IMG_FILE" && TEST_IMG_FILE=$TEST_IMG

size=128M
_make_test_img $size

echo
echo "== mark image read-only"
chmod a-w "$TEST_IMG_FILE"

echo
echo "== read from read-only image"
$QEMU_IO -r -c "read 0 512" "$TEST_IMG" | _filter_qemu_io

# success, all done
echo "*** done"
rm -f $seq.full
status=0
