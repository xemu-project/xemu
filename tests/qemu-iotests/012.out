QA output created by 012
Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=134217728

== mark image read-only

== read from read-only image
read 512/512 bytes at offset 0
512 bytes, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)
*** done
